### ISSUE TITLE
jupyter reports "maximum recursion depth exceeded in comparison"

### ISSUE THREAD

**Issue by xushijie** [microsoft/guidance#90]
**The bug**

The jupyter reports 
 `maximum recursion depth exceeded in comparison` 
exception when execute program 

```Traceback (most recent call last):
  File "/work/python/env/lib/python3.9/site-packages/guidance/_program_executor.py", line 94, in run
    await self.visit(self.parse_tree)
  File "/work/python/env/lib/python3.9/site-packages/guidance/_program_executor.py", line 428, in visit
    visited_children.append(await self.visit(child, inner_next_node, inner_next_next_node, inner_prev_node, node, parent_node))
  File "/work/python/env/lib/python3.9/site-packages/guidance/_program_executor.py", line 428, in visit
    visited_children.append(await self.visit(child, inner_next_node, inner_next_next_node, inner_prev_node, node, parent_node))
  File "/work/python/env/lib/python3.9/site-packages/guidance/_program_executor.py", line 395, in visit
    command_output = await command_function(*positional_args, **named_args)
  File "/work/python/env/lib/python3.9/site-packages/guidance/library/_select.py", line 165, in select
    option_logprobs = await recursive_select("")
  File "/work/python/env/lib/python3.9/site-packages/guidance/library/_select.py", line 140, in recursive_select
    sub_logprobs = await recursive_select(rec_prefix)
  File "/work/python/env/lib/python3.9/site-packages/guidance/library/_select.py", line 140, in recursive_select
    sub_logprobs = await recursive_select(rec_prefix)
  File "/work/python/env/lib/python3.9/site-packages/guidance/library/_select.py", line 140, in recursive_select
    sub_logprobs = await recursive_select(rec_prefix)
  [Previous line repeated 2946 more times]
  File "/work/python/env/lib/python3.9/site-packages/guidance/library/_select.py", line 110, in recursive_select
    gen_obj = await parser.llm_session(
  File "/work/python/env/lib/python3.9/site-packages/guidance/llms/_openai.py", line 485, in __call__
    return self.llm.__class__.cache[key]
  File "/work/python/env/lib/python3.9/site-packages/diskcache/core.py", line 1232, in __getitem__
    value = self.get(key, default=ENOVAL, retry=True)
  File "/work/python/env/lib/python3.9/site-packages/diskcache/core.py", line 1173, in get
    value = self._disk.fetch(mode, filename, db_value, read)
  File "/work/python/env/lib/python3.9/site-packages/diskcache/core.py", line 284, in fetch
    return pickle.load(io.BytesIO(value))
  File "/opt/anaconda/lib/python3.9/site-packages/openai/openai_object.py", line 30, in __init__
    self._response_ms = response_ms
  File "/opt/anaconda/lib/python3.9/site-packages/openai/openai_object.py", line 49, in __setattr__
    if k[0] == "_" or k in self.__dict__:
RecursionError: maximum recursion depth exceeded in comparison

Error in program:  maximum recursion depth exceeded in comparison
---------------------------------------------------------------------------
RecursionError                            Traceback (most recent call last)
Input In [45], in <cell line: 10>()
      2 # the {{#select}} command allows you to use the LLM to select from a set of options
      3 program = guidance('''Is the following sentence offensive? Please answer with a single word, either "Yes", "No", or "Maybe".
      4 Sentence: {{example}}
      5 Answer:{{#select "answer" logprobs='logprobs'}} 
   (...)
      8      Maybe
      9 {{/select}}''')
---> 10 executed_program = program(example='I hate tacos')

File /work/python/env/lib/python3.9/site-packages/guidance/_program.py:226, in Program.__call__(self, **kwargs)
    224     loop = asyncio.new_event_loop()
    225     loop.create_task(new_program.update_display.run()) # start the display updater
--> 226     loop.run_until_complete(new_program.execute())
    228 return new_program

File /opt/anaconda/lib/python3.9/site-packages/nest_asyncio.py:89, in _patch_loop.<locals>.run_until_complete(self, future)
     86 if not f.done():
     87     raise RuntimeError(
     88         'Event loop stopped before Future completed.')
---> 89 return f.result()

File /opt/anaconda/lib/python3.9/asyncio/futures.py:201, in Future.result(self)
    199 self.__log_traceback = False
    200 if self._exception is not None:
--> 201     raise self._exception
    202 return self._result

File /opt/anaconda/lib/python3.9/asyncio/tasks.py:256, in Task.__step(***failed resolving arguments***)
    252 try:
    253     if exc is None:
    254         # We use the `send` method directly, because coroutines
    255         # don't have `__iter__` and `__next__` methods.
--> 256         result = coro.send(None)
    257     else:
    258         result = coro.throw(exc)

File /work/python/env/lib/python3.9/site-packages/guidance/_program.py:312, in Program.execute(self)
    310 else:
    311     with self.llm.session(asynchronous=True) as llm_session:
--> 312         await self._executor.run(llm_session)
    313 self._text = self._executor.prefix
    315 # delete the executor and so mark the program as not executing

File /work/python/env/lib/python3.9/site-packages/guidance/_program_executor.py:98, in ProgramExecutor.run(self, llm_session)
     96 print(traceback.format_exc())
     97 print("Error in program: ", e)
---> 98 raise e

File /work/python/env/lib/python3.9/site-packages/guidance/_program_executor.py:94, in ProgramExecutor.run(self, llm_session)
     88 self.llm_session = llm_session
     89 try:
     90     # first parse all the whitespace control
     91     # self.whitespace_control_visit(self.parse_tree)
     92 
     93     # now execute the program
---> 94     await self.visit(self.parse_tree)
     95 except Exception as e:
     96     print(traceback.format_exc())

File /work/python/env/lib/python3.9/site-packages/guidance/_program_executor.py:428, in ProgramExecutor.visit(self, node, next_node, next_next_node, prev_node, parent_node, grandparent_node)
    426     else:
    427         inner_prev_node = prev_node
--> 428     visited_children.append(await self.visit(child, inner_next_node, inner_next_next_node, inner_prev_node, node, parent_node))
    429 # visited_children = [self.visit(child) for child in node.children]
    431 if len(visited_children) == 1:

File /work/python/env/lib/python3.9/site-packages/guidance/_program_executor.py:428, in ProgramExecutor.visit(self, node, next_node, next_next_node, prev_node, parent_node, grandparent_node)
    426     else:
    427         inner_prev_node = prev_node
--> 428     visited_children.append(await self.visit(child, inner_next_node, inner_next_next_node, inner_prev_node, node, parent_node))
    429 # visited_children = [self.visit(child) for child in node.children]
    431 if len(visited_children) == 1:

File /work/python/env/lib/python3.9/site-packages/guidance/_program_executor.py:395, in ProgramExecutor.visit(self, node, next_node, next_next_node, prev_node, parent_node, grandparent_node)
    393 # call the optionally asyncronous command
    394 if inspect.iscoroutinefunction(command_function):
--> 395     command_output = await command_function(*positional_args, **named_args)
    396 else:
    397     command_output = command_function(*positional_args, **named_args)

File /work/python/env/lib/python3.9/site-packages/guidance/library/_select.py:165, in select(variable_name, options, logprobs, list_append, _parser_context)
    162     return logprobs_out
    164 # recursively compute the logprobs for each option
--> 165 option_logprobs = await recursive_select("")
    167 selected_option = max(option_logprobs, key=option_logprobs.get)
    169 # see if we are appending to a list or not

File /work/python/env/lib/python3.9/site-packages/guidance/library/_select.py:140, in select.<locals>.recursive_select(current_prefix, allow_token_extension)
    138     sub_logprobs = await recursive_select(rec_prefix, allow_token_extension=False)
    139 else:
--> 140     sub_logprobs = await recursive_select(rec_prefix)
    142 # we add the logprob of this token to the logprob of the suffix
    143 for k in sub_logprobs:

File /work/python/env/lib/python3.9/site-packages/guidance/library/_select.py:140, in select.<locals>.recursive_select(current_prefix, allow_token_extension)
    138     sub_logprobs = await recursive_select(rec_prefix, allow_token_extension=False)
    139 else:
--> 140     sub_logprobs = await recursive_select(rec_prefix)
    142 # we add the logprob of this token to the logprob of the suffix
    143 for k in sub_logprobs:

    [... skipping similar frames: select.<locals>.recursive_select at line 140 (2946 times)]

File /work/python/env/lib/python3.9/site-packages/guidance/library/_select.py:140, in select.<locals>.recursive_select(current_prefix, allow_token_extension)
    138     sub_logprobs = await recursive_select(rec_prefix, allow_token_extension=False)
    139 else:
--> 140     sub_logprobs = await recursive_select(rec_prefix)
    142 # we add the logprob of this token to the logprob of the suffix
    143 for k in sub_logprobs:

File /work/python/env/lib/python3.9/site-packages/guidance/library/_select.py:110, in select.<locals>.recursive_select(current_prefix, allow_token_extension)
    107     last_token_str = ""
    109 # generate the token logprobs
--> 110 gen_obj = await parser.llm_session(
    111     call_prefix, # TODO: perhaps we should allow passing of token ids directly? (this could allow us to avoid retokenizing the whole prefix many times)
    112     max_tokens=1,
    113     logit_bias=logit_bias,
    114     logprobs=10,
    115     cache_seed=0,
    116     token_healing=False # we manage token boundary healing ourselves for this function
    117 )
    118 logprobs_result = gen_obj["choices"][0]["logprobs"]
    119 top_logprobs = logprobs_result["top_logprobs"][0]

File /work/python/env/lib/python3.9/site-packages/guidance/llms/_openai.py:485, in OpenAISession.__call__(self, prompt, stop, stop_regex, temperature, n, max_tokens, logprobs, top_p, echo, logit_bias, token_healing, pattern, stream, cache_seed, caching)
    482         return self.llm.__class__.cache[key]
    483     return [self.llm.__class__.cache[key]]
--> 485 return self.llm.__class__.cache[key]

File /work/python/env/lib/python3.9/site-packages/diskcache/core.py:1232, in Cache.__getitem__(self, key)
   1224 def __getitem__(self, key):
   1225     """Return corresponding value for `key` from cache.
   1226 
   1227     :param key: key matching item
   (...)
   1230 
   1231     """
-> 1232     value = self.get(key, default=ENOVAL, retry=True)
   1233     if value is ENOVAL:
   1234         raise KeyError(key)

File /work/python/env/lib/python3.9/site-packages/diskcache/core.py:1173, in Cache.get(self, key, default, read, expire_time, tag, retry)
   1170 ((rowid, db_expire_time, db_tag, mode, filename, db_value),) = rows
   1172 try:
-> 1173     value = self._disk.fetch(mode, filename, db_value, read)
   1174 except IOError:
   1175     # Key was deleted before we could retrieve result.
   1176     return default

File /work/python/env/lib/python3.9/site-packages/diskcache/core.py:284, in Disk.fetch(self, mode, filename, value, read)
    282         return pickle.load(reader)
    283 else:
--> 284     return pickle.load(io.BytesIO(value))

File /opt/anaconda/lib/python3.9/site-packages/openai/openai_object.py:30, in OpenAIObject.__init__(self, id, api_key, api_version, api_type, organization, response_ms, api_base, engine, **params)
     28 if response_ms is not None and not isinstance(response_ms, int):
     29     raise TypeError(f"response_ms is a {type(response_ms).__name__}.")
---> 30 self._response_ms = response_ms
     32 self._retrieve_params = params
     34 object.__setattr__(self, "api_key", api_key)

File /opt/anaconda/lib/python3.9/site-packages/openai/openai_object.py:49, in OpenAIObject.__setattr__(self, k, v)
     48 def __setattr__(self, k, v):
---> 49     if k[0] == "_" or k in self.__dict__:
     50         return super(OpenAIObject, self).__setattr__(k, v)
     52     self[k] = v

RecursionError: maximum recursion depth exceeded in comparison
```

**To Reproduce**
I copied a sample code from tutorial 

```python
guidance.llm = guidance.llms.OpenAI("text-davinci-003")
# the {{#select}} command allows you to use the LLM to select from a set of options
program = guidance('''Is the following sentence offensive? Please answer with a single word, either "Yes", "No", or "Maybe".
Sentence: {{example}}
Answer:{{#select "answer" logprobs='logprobs'}} 
     Yes{{or}} 
     No{{or}} 
     Maybe
{{/select}}''')
executed_program = program(example='I hate tacos')
```

**System info (please complete the following information):**
 - OS (e.g. Ubuntu, Windows 11, Mac OS, etc.): Ubuntu 2204
 - Guidance Version (`guidance.__version__`):  '0.0.51'

**Comment by slundberg** [comment]
Thanks, this has to do with some odd tokenization effects of long strings of spaces. Will push a fix soon.

**Comment by slundberg** [comment]
Closing. Now fixed in 0.0.54
